pyfar/classes/filter.py

```diff
@@ -389,7 +389,22 @@ def _process(coefficients, data, zi=None):
         This is a hidden static method required for a shared processing
         function in the parent class.
         """
-        return spsignal.lfilter(coefficients[0], 1, data, zi=zi)
+        if zi is None:
+            # dynamically add new dimensions to b to match ndims of data
+            # for convolution
+            b = coefficients[0]
+            new_dims = data.ndim - b.ndim
+
+            for _i in range(new_dims):
+                b = b[np.newaxis, ...]
```

Comment:
This should work as well and is shorter
```suggestion
            b = np.broadcast_to(b, (1, ) * (data.ndim - b.ndim) + b.shape)
```

Maybe `data.ndim - b.ndim` is not even needed, because we allways pas a single channel. `data.dnim - 1` might work as well.